File: Ruby/005.rb
def gcd(a, b)
  while b != 0 do
    temp = b
    b = a % b
    a = temp
  end
  return a
end

a = 1
lcm = 1
2.upto(20).each do |b|
  lcm = (a * b) / gcd(a, b)
  a = lcm
end
puts lcm

File: Cpp/005.cpp
#include<cstdio>
#include<iostream>
using namespace std;
typedef long long ll;

ll gcd(ll a,ll b)
{
    while(b!=0)
    {
        ll temp=b;
        b=a%b;
        a=temp;
    }
    return a;
}

int main()
{
    ll a=1;
    ll lcm;
    for(ll b=2;b<=20;b++)
    {
        lcm=a*b/gcd(a,b);
        a=lcm;
    }
    cout<<lcm;
}
